packages/fxa-settings/src/pages/ResetPassword/index.stories.tsx

```diff
@@ -16,41 +16,59 @@ import {
   mockDefaultAccount,
 } from './mocks';
 
+import {
+  produceComponent,
+  createAppContext,
+  createHistoryWithQuery,
+} from '../../models/mocks';
+
 export default {
   title: 'Pages/ResetPassword',
   component: ResetPassword,
   decorators: [withLocalization],
 } as Meta;
 
-const storyWithProps = (
+const route = '/reset_password';
+
+function render(
   account: Account,
-  props?: Partial<ResetPasswordProps>
-) => {
-  const story = () => (
-    <AppContext.Provider value={{ account }}>
-      <LocationProvider>
-        <ResetPassword {...props} />
-      </LocationProvider>
-    </AppContext.Provider>
+  props?: Partial<ResetPasswordProps>,
+  queryParams?: string
+) {
+  const history = createHistoryWithQuery(route, queryParams);
+  return produceComponent(
+    <ResetPassword {...props} />,
+    { route, history },
+    {
+      ...createAppContext(history),
+      account,
+    }
   );
-  return story;
-};
+}
 
-export const Default = storyWithProps(mockDefaultAccount);
+export const Default = () => {
+  return render(mockDefaultAccount);
+};
 
-export const WithServiceName = storyWithProps(mockDefaultAccount, {
-  serviceName: MozServices.MozillaVPN,
-});
+export const WithServiceName = () => {
+  return render(
+    mockDefaultAccount,
+    undefined,
+    `service=${MozServices.MozillaVPN}`
```

Comment:
We could have a comment that in reality this would be an ID, or maybe add a comment to some related ticket (like the Moz service enums ticket), or maybe we need a new one to address [this comment](https://github.com/LZoog/fxa/blob/d66211693dfb4773d859e74dceb139daf812ecee/packages/fxa-settings/src/components/CardHeader/index.tsx#L125-L126) where it could be looked at there.

Reply:
I'll file a followup.